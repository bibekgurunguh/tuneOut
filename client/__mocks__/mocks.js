const mocks = {};

mocks.mockSongInfo = {
  status: {
    msg: 'Success',
    code: 0,
    version: '1.0',
  },
  metadata: {
    music: [
      {
        external_ids: {},
        contributors: { composers: ['Andr√© Boadu'] },
        label: 'Keinemusik',
        duration_ms: 477360,
        score: 100,
        play_offset_ms: 197640,
        release_date: '2015-07-10',
        genres: [{ name: 'Dance' }],
        external_metadata: {},
        title: 'Woods',
        acrid: '5c748ee8fc84dc80ef38cee6580cc890',
        album: { name: 'Trilogy' },
        result_from: 3,
        artists: [{ name: '&ME' }],
      },
    ],
    timestamp_utc: '2021-01-18 18:35:18',
  },

  cost_time: 0.24199986457825,
  result_type: 0,
};

module.exports = mocks;
