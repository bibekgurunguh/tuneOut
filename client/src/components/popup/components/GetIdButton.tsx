/* global chrome*/
import * as React from 'react';
import './GetIdButton.css';

interface PropsType {
  getId: Function,
  runLoadingAnimation: Function
}

export function GetIdButton({
  getId,
  runLoadingAnimation,
}: PropsType) {
  function handleClick() {
    runLoadingAnimation();
    getId();
  }

  return (
    <div>
      <button className={'button'} onClick={handleClick}>
        identify
      </button>
    </div>
  );
}
