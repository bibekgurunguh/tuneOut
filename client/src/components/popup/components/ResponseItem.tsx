import * as React from 'react';
import './ResponseItem.css'

interface PropsType {
  item: string,
  attribute: string,
}

export default function ResponseItem({ item, attribute }: PropsType) {

  return (
    <div>
      <h3 className={'responseItem'}><strong className={'attribute'}>{attribute}</strong>{item}</h3>
    </div>
  )
}
