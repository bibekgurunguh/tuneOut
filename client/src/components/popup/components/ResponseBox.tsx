import * as React from 'react';
import ResponseItem from './ResponseItem';
import './ResponseBox.css';
import Lottie from 'react-lottie';
// import animationData from '../animations/loading-animation.json'
const animationData = require('../animations/loading-animation.json');
import YtSearch from './external-links/YtSearch';
import SpotifySearch from './external-links/SpotifySearch';
import DiscogsSearch from './external-links/DiscogsSearch';

export function ResponseBox({ songInfo, setSongInfo, animation }) {
  const defaultOptions = {
    loop: true,
    autoplay: true,
    animationData: animationData,
    rendererSettings: {
      preserveAspectRatio: 'xMidYMid slice',
    },
  };

  if (animation === true) {
    return (
      <div className={'animation'}>
        <Lottie options={defaultOptions} height={260} width={275} />
      </div>
    );
  }

  if (songInfo) {
    const songObj = JSON.parse(songInfo);
    if (songObj.status.code === 1001) {
      //? on song identification failure
      songInfo = [];
      return (
        <div className={'errorBox'}>
          <h3 className={'errorMessage'}>couldn't identify track</h3>
          <h3 className={'errorMessage'}>please try again</h3>
        </div>
      );
    } else if (songObj.status.code === 2004) {
      //? on song fingerprint failure
      songInfo = [];
      return (
        <div className={'errorBox'}>
          <h3 className={'errorMessage'}>couldn't detect audio</h3>
          <h3 className={'errorMessage'}>please try again</h3>
        </div>
      );
    }

    // console.log('parser', songObj)
    // console.log('typeofparsed', typeof songObj)
    // console.log('songinfo.metadata', songObj.metadata)

    const id = songObj.metadata.music[0];

    const title = id.title;
    const artist = id.artists[0].name;
    const label = id.label;
    const released = id.release_date;
    const album = id.album.name;

    const trackId = 'Track ID: ';
    const trackArtist = 'Artist: ';
    const trackReleased = 'Released: ';
    const trackAlbum = 'Album: ';
    const trackLabel = 'Label: ';

    let externalIdSpotify;
    let externalIdYoutube;

    if (id.external_metadata.spotify) {
      externalIdSpotify = id.external_metadata.spotify.track.id;
    }

    if (id.external_metadata.youtube) {
      externalIdYoutube = id.external_metadata.youtube.vid;
    }

    //? on song identification successful
    return (
      <div className={'resBox'}>
        <ResponseItem attribute={trackId} item={title}></ResponseItem>
        <ResponseItem attribute={trackArtist} item={artist}></ResponseItem>
        <ResponseItem attribute={trackReleased} item={released}></ResponseItem>
        <ResponseItem attribute={trackAlbum} item={album}></ResponseItem>
        <ResponseItem attribute={trackLabel} item={label}></ResponseItem>
        <div className={'linkbox'}>
          <div className="yt">
            <YtSearch
              externalIdYoutube={externalIdYoutube}
              title={title}
              artist={artist}
            ></YtSearch>
          </div>
          <div className="spotify">
            <SpotifySearch
              externalIdSpotify={externalIdSpotify}
              title={title}
              artist={artist}
            ></SpotifySearch>
          </div>
          <DiscogsSearch artist={artist} album={album}></DiscogsSearch>
        </div>
      </div>
    );
  }

  //? On popup initialization
  const firstMsg = 'navigate to an audible tab';
  const secondMsg = 'click identify to start';

  return (
    <div className={'emptyResBox'}>
      <ResponseItem attribute={firstMsg} item={''}></ResponseItem>
      <ResponseItem attribute={secondMsg} item={''}></ResponseItem>
    </div>
  );
}
