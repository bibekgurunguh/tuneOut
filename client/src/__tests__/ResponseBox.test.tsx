import React from 'react';
import {
  render,
  screen,
  fireEvent,
  waitForElement,
} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import '@testing-library/jest-dom';
import 'jest-canvas-mock';

import { mockSongInfo } from '../../__mocks__/mocks';

import ResponseItem from '../components/popup/components/ResponseItem';
import { ResponseBox } from '../components/popup/components/ResponseBox';

describe('<ResponseBox /> should render various elements of the song result correctly', () => {
  beforeEach(() => {
    render(
      <ResponseBox
        animation={false}
        setSongInfo={''}
        songInfo={JSON.stringify(mockSongInfo)}
      ></ResponseBox>,
    );
  });
  it('should render the trackID correctly', () => {
    const trackId = screen.getByText('Woods');
    expect(trackId).toBeInTheDocument();
  });
  it('should render the Artist correctly', () => {
    const trackArtist = screen.getByText('&ME');
    expect(trackArtist).toBeInTheDocument();
  });
  it('should render the release date correctly', () => {
    const trackReleased = screen.getByText('2015-07-10');
    expect(trackReleased).toBeInTheDocument();
  });
  it('should render the album title correctly', () => {
    const albumTitle = screen.getByText('Trilogy');
    expect(albumTitle).toBeInTheDocument();
  });
  it('should render the Label correctly', () => {
    const artistLabel = screen.getByText('Keinemusik');
    expect(artistLabel).toBeInTheDocument();
  });
  it('should render all elements together', () => {
    const trackId = screen.getByText('Woods');
    const trackArtist = screen.getByText('&ME');
    const trackReleased = screen.getByText('2015-07-10');
    const albumTitle = screen.getByText('Trilogy');
    const artistLabel = screen.getByText('Keinemusik');

    expect(trackId).toBeInTheDocument();
    expect(trackArtist).toBeInTheDocument();
    expect(trackReleased).toBeInTheDocument();
    expect(albumTitle).toBeInTheDocument();
    expect(artistLabel).toBeInTheDocument();
  });
  test('rendered ResponseBox matches the existing snapshot when songInfo is provided', () => {
    const tree = render(
      <ResponseBox
        songInfo={JSON.stringify(mockSongInfo)}
        setSongInfo=""
        animation="false"
      />,
    );
    expect(tree).toMatchSnapshot();
  });
});

describe('<ResponseItem />', () => {
  test('should render responseItem to match previous snapshot', () => {
    const tree = render(<ResponseItem item="trackId" attribute="Woods" />);
    expect(tree).toMatchSnapshot();
  });
});
