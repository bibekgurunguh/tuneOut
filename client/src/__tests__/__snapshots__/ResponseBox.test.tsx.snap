// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResponseBox /> should render various elements of the song result correctly rendered ResponseBox matches the existing snapshot when songInfo is provided 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="resBox"
      >
        <div>
          <h3
            class="responseItem"
          >
            <strong
              class="attribute"
            >
              Track ID: 
            </strong>
            Woods
          </h3>
        </div>
        <div>
          <h3
            class="responseItem"
          >
            <strong
              class="attribute"
            >
              Artist: 
            </strong>
            &ME
          </h3>
        </div>
        <div>
          <h3
            class="responseItem"
          >
            <strong
              class="attribute"
            >
              Released: 
            </strong>
            2015-07-10
          </h3>
        </div>
        <div>
          <h3
            class="responseItem"
          >
            <strong
              class="attribute"
            >
              Album: 
            </strong>
            Trilogy
          </h3>
        </div>
        <div>
          <h3
            class="responseItem"
          >
            <strong
              class="attribute"
            >
              Label: 
            </strong>
            Keinemusik
          </h3>
        </div>
        <div
          class="linkbox"
        >
          <div
            class="yt"
          >
            <div>
              <a
                href="https://www.youtube.com/results?search_query=Woods+&ME"
                target="_blank"
              >
                <img
                  alt="Search on Youtube"
                  class="ytlogo"
                  src="test-file-stub"
                />
              </a>
            </div>
          </div>
          <div
            class="spotify"
          >
            <div>
              <a
                href="https://open.spotify.com/search/&ME%20Woods"
                target="_blank"
              >
                <img
                  alt="Search on Spotify"
                  class="spotifylogo"
                  src="test-file-stub"
                />
              </a>
            </div>
          </div>
          <div>
            <a
              href="https://www.discogs.com/search/?q=&ME+Trilogy&type=all"
              target="_blank"
            >
              <img
                alt="Search on Discogs"
                src="test-file-stub"
              />
            </a>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        class="resBox"
      >
        <div>
          <h3
            class="responseItem"
          >
            <strong
              class="attribute"
            >
              Track ID: 
            </strong>
            Woods
          </h3>
        </div>
        <div>
          <h3
            class="responseItem"
          >
            <strong
              class="attribute"
            >
              Artist: 
            </strong>
            &ME
          </h3>
        </div>
        <div>
          <h3
            class="responseItem"
          >
            <strong
              class="attribute"
            >
              Released: 
            </strong>
            2015-07-10
          </h3>
        </div>
        <div>
          <h3
            class="responseItem"
          >
            <strong
              class="attribute"
            >
              Album: 
            </strong>
            Trilogy
          </h3>
        </div>
        <div>
          <h3
            class="responseItem"
          >
            <strong
              class="attribute"
            >
              Label: 
            </strong>
            Keinemusik
          </h3>
        </div>
        <div
          class="linkbox"
        >
          <div
            class="yt"
          >
            <div>
              <a
                href="https://www.youtube.com/results?search_query=Woods+&ME"
                target="_blank"
              >
                <img
                  alt="Search on Youtube"
                  class="ytlogo"
                  src="test-file-stub"
                />
              </a>
            </div>
          </div>
          <div
            class="spotify"
          >
            <div>
              <a
                href="https://open.spotify.com/search/&ME%20Woods"
                target="_blank"
              >
                <img
                  alt="Search on Spotify"
                  class="spotifylogo"
                  src="test-file-stub"
                />
              </a>
            </div>
          </div>
          <div>
            <a
              href="https://www.discogs.com/search/?q=&ME+Trilogy&type=all"
              target="_blank"
            >
              <img
                alt="Search on Discogs"
                src="test-file-stub"
              />
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="resBox"
    >
      <div>
        <h3
          class="responseItem"
        >
          <strong
            class="attribute"
          >
            Track ID: 
          </strong>
          Woods
        </h3>
      </div>
      <div>
        <h3
          class="responseItem"
        >
          <strong
            class="attribute"
          >
            Artist: 
          </strong>
          &ME
        </h3>
      </div>
      <div>
        <h3
          class="responseItem"
        >
          <strong
            class="attribute"
          >
            Released: 
          </strong>
          2015-07-10
        </h3>
      </div>
      <div>
        <h3
          class="responseItem"
        >
          <strong
            class="attribute"
          >
            Album: 
          </strong>
          Trilogy
        </h3>
      </div>
      <div>
        <h3
          class="responseItem"
        >
          <strong
            class="attribute"
          >
            Label: 
          </strong>
          Keinemusik
        </h3>
      </div>
      <div
        class="linkbox"
      >
        <div
          class="yt"
        >
          <div>
            <a
              href="https://www.youtube.com/results?search_query=Woods+&ME"
              target="_blank"
            >
              <img
                alt="Search on Youtube"
                class="ytlogo"
                src="test-file-stub"
              />
            </a>
          </div>
        </div>
        <div
          class="spotify"
        >
          <div>
            <a
              href="https://open.spotify.com/search/&ME%20Woods"
              target="_blank"
            >
              <img
                alt="Search on Spotify"
                class="spotifylogo"
                src="test-file-stub"
              />
            </a>
          </div>
        </div>
        <div>
          <a
            href="https://www.discogs.com/search/?q=&ME+Trilogy&type=all"
            target="_blank"
          >
            <img
              alt="Search on Discogs"
              src="test-file-stub"
            />
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<ResponseItem /> should render responseItem to match previous snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h3
          class="responseItem"
        >
          <strong
            class="attribute"
          >
            Woods
          </strong>
          trackId
        </h3>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h3
        class="responseItem"
      >
        <strong
          class="attribute"
        >
          Woods
        </strong>
        trackId
      </h3>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
